src/report/report.service.ts

```diff
@@ -30,27 +33,33 @@ export class ReportService {
     return ReportService.docToPublicReport(report);
   }
 
-  async getReportStatistics(): Promise<ReportStatisticsDto> {
-    const stats = await this.reportRepository.getVisibleStatusCounts();
+  async getReportStatistics(
+    category?: ReportCategory,
+  ): Promise<ReportStatisticsDto> {
+    const stats = await this.reportRepository.getVisibleStatusCounts(category);
     return ReportService.docToReportStatistics(stats);
   }
 
   private static docToReportStatistics(e: any): ReportStatisticsDto {
     return new ReportStatisticsDto(
-      e.filter((stat: { _id: string }) => stat._id == 'gautas')[0].count ?? 0,
-      e.filter((stat: { _id: string }) => stat._id == 'tiriamas')[0].count ?? 0,
-      e.filter((stat: { _id: string }) => stat._id == 'ištirtas')[0].count ?? 0,
-      e.filter((stat: { _id: string }) => stat._id == 'sutvarkyta')[0].count ??
-        0,
-      e.filter((stat: { _id: string }) => stat._id == 'nepasitvirtino')[0]
-        .count ?? 0,
+      ReportService.filterStatistics(e, 'gautas'),
```

Comment:
Please note that this change appears unrelated to the current pull request. For future reference, it's best to create a new pull request specifically for this bug fix.